pkg/engine/datasource/kafka_datasource/kafka_consumer_group.go

```diff
@@ -123,14 +124,53 @@ func (k *KafkaConsumerGroup) startConsuming(handler sarama.ConsumerGroupHandler)
 		}
 	}()
 
-	// Blocking call
-	err := k.consumerGroup.Consume(k.ctx, []string{k.options.Topic}, handler)
-	if err != nil {
-		k.log.Error("KafkaConsumerGroup.startConsuming",
+	k.wg.Add(1)
+	go func() {
+		defer k.wg.Done()
+
+		// Errors returns a read channel of errors that occurred during the consumer life-cycle.
+		// By default, errors are logged and not returned over this channel.
+		// If you want to implement any custom error handling, set your config's
+		// Consumer.Return.Errors setting to true, and read from this channel.
+		for err := range k.consumerGroup.Errors() {
+			k.log.Error("KafkaConsumerGroup.Consumer",
+				log.String("topic", k.options.Topic),
+				log.String("groupID", k.options.GroupID),
+				log.String("clientID", k.options.ClientID),
+				log.Error(err))
+		}
+	}()
+
+	// From Sarama documents:
+	//
+	// This method should be called inside an infinite loop, when a
+	// server-side rebalance happens, the consumer session will need to be
+	// recreated to get the new claims.
+	for {
+		select {
+		case <-k.ctx.Done():
+			return
+		default:
+		}
+
+		k.log.Info("KafkaConsumerGroup.consumerGroup.Consume has been called",
 			log.String("topic", k.options.Topic),
 			log.String("groupID", k.options.GroupID),
-			log.String("clientID", k.options.ClientID),
-			log.Error(err))
+			log.String("clientID", k.options.ClientID))
+
+		// Blocking call
+		err := k.consumerGroup.Consume(k.ctx, []string{k.options.Topic}, handler)
+		if err != nil {
```

Comment:
Obviously not an issue, but would it be cleaner to wrap the if on the same line?

Reply:
I usually prefer checking the error on a different line. Because it's mostly easy to read.